Add teleport and escape packet

Zeroeh opened the pull request.

Zeroeh committed:
Add teleport and escape packet

toddw123 reviewed (changes requested):
I cant approve this since it changes some of the project config files. It might be best to just add the project config files to the gitignore i guess. 
Also i plan to add objects.xml + tiles.xml to the project. From what i can tell, the gitignore change you are making will exclude those. 
Escape and teleport look great though

Zeroeh committed:
Update .gitignore

toddw123 commented:
I see you removed the *.xml line from the gitignore, but i suggest you do a pull and get the recent gitignore i just pushed. It just flat out ignores any files named "settings.xml" so you can put it wherever you need. It should also be ignoring any project config changes now, i hope.

Zeroeh committed:
Update

Zeroeh committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	.gitignore

Zeroeh committed:
#ifdef DEBUG_OUTPUT

Let the user decide if they want all this output, it gets really old
after a while
